flexmeasures/ui/crud/assets/services.py

```diff
@@ -0,0 +1,261 @@
+            data["latitude"] = float(data["latitude"])
+
+        if "csrf_token" in data:
+            del data["csrf_token"]
+
+        return data
+
+    def process_api_validation_errors(self, api_response: dict):
+        """Process form errors from the API for the WTForm"""
+        if not isinstance(api_response, dict):
+            return
+        for error_header in ("json", "validation_errors"):
+            if error_header not in api_response:
+                continue
+            for field in list(self._fields.keys()):
+                if field in list(api_response[error_header].keys()):
+                    field_errors = api_response[error_header][field]
+                    if isinstance(field_errors, list):
+                        self._fields[field].errors += api_response[error_header][field]
+                    else:
+                        self._fields[field].errors.append(
+                            api_response[error_header][field]
+                        )
+
+    def with_options(self):
+        if "generic_asset_type_id" in self:
+            self.generic_asset_type_id.choices = [(-1, "--Select type--")] + [
+                (atype.id, atype.name)
+                for atype in db.session.scalars(select(GenericAssetType)).all()
+            ]
+        if "account_id" in self:
+            self.account_id.choices = [(-1, "--Select account--")] + [
+                (account.id, account.name)
+                for account in db.session.scalars(select(Account)).all()
+            ]
+
+    def get_allowed_price_sensor_data(
+        self, account_id: Optional[int]
+    ) -> Dict[int, str]:
+        """
+        Return a list of sensors which the user can add
+        as consumption_price_sensor_id or production_price_sensor_id.
+        For each sensor we get data as sensor_id: asset_name:sensor_name.
+        """
+        if not account_id:
+            assets = db.session.scalars(
+                select(GenericAsset).filter(GenericAsset.account_id.is_(None))
+            ).all()
+        else:
+            assets = db.session.scalars(
+                select(GenericAsset).filter(
+                    or_(
+                        GenericAsset.account_id == account_id,
+                        GenericAsset.account_id.is_(None),
+                    )
+                )
+            ).all()
+
+        sensors_data = list()
+        for asset in assets:
+            sensors_data += [
+                (sensor.id, asset.name, sensor.name, sensor.unit)
+                for sensor in asset.sensors
+            ]
+
+        return {
+            sensor_id: f"{asset_name}:{sensor_name}"
+            for sensor_id, asset_name, sensor_name, sensor_unit in sensors_data
+            if is_energy_price_unit(sensor_unit)
+        }
+
+    def get_allowed_inflexible_sensor_data(
+        self, account_id: Optional[int]
+    ) -> Dict[int, str]:
+        """
+        Return a list of sensors which the user can add
```

Comment:
Please mention that we select sensors with energy units within the current account (or among public assets)